api/test/deals.go

```diff
@@ -43,21 +43,42 @@ func TestDealFlow(t *testing.T, b APIBuilder, blocktime time.Duration) {
 	}
 	time.Sleep(time.Second)
 
-	data := make([]byte, 600)
-	rand.New(rand.NewSource(5)).Read(data)
+	mine := true
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		for mine {
+			time.Sleep(blocktime)
+			if err := sn[0].MineOne(ctx); err != nil {
+				t.Error(err)
+			}
+		}
+	}()
 
-	r := bytes.NewReader(data)
-	fcid, err := client.ClientImportLocal(ctx, r)
+	makeDeal(t, ctx, 6, client, miner)
```

Comment:
So most of the deal making is now on an extracted `makeDeal`.
This makes `TestDealFlow` only calling this once, and `TestDoubleDealFlow` calling it twice... to be sure both are doing the exact same thing (with just randomness changing for the second deal).